pkg/cli/cmd/resource/list/list.go

```diff
@@ -58,15 +58,28 @@ rad resource list Applications.Core/containers --application icecream-store
 
 # list all resources of a specified type in an application (shorthand flag)
 rad resource list Applications.Core/containers -a icecream-store
+
+# list all resources of a specified type in an environment
+rad resource list Applications.Core/containers --environment dev-env
+
+# list all resources of a specified type in an environment (shorthand flag)
+rad resource list Applications.Core/containers -e dev-env
+
+# list all resources in an environment (no resource type specified)
+rad resource list --environment dev-env
+
+# list all resources in the default environment
+rad resource list
```

Comment:
Would it make sense to return all resources in all environments if this is the case? This is not in the scope of this PR but just wanted to brainstorm on this potential feature. cc/ @willtsai

Reply:
no, I think that listing the resources in the default environment is good since using the default if none provided is a pattern that's applied across the other commands.

Reply:
Is it possible to display the name of the environment when rendering the results? This would be especially useful in the case where we fall back to a default when user doesn't provide an environment flag

Reply:
@willtsai - Is this something you want implemented in this PR? If so, am I right in saying that the changes would need to be made in `objectformats`? @ytimocin

Reply:
Yes, if possible it would be great to include it in this PR. I'll get @ytimocin to confirm for you.

Reply:
@willtsai & @ytimocin  - I've pushed a change to display the environment name when rendering `rad resource list` - Please let me know if I've got it right or need to make changes :)